src/vendor/event/mod.rs

```diff
@@ -3291,3 +3002,163 @@ fn to_hal_firmware_error(buffer: &[u8]) -> Result<HalFirmwareError, crate::event
         data,
     })
 }
+
+#[derive(Debug, Clone, Copy)]
+#[cfg_attr(feature = "defmt", derive(defmt::Format))]
+pub struct CocConnectEvent {
+        /// The connection handle related to the event.
+        pub conn_handle: ConnectionHandle,
+
+        pub smps: u16,
+        pub mtu: u16,
+        pub mps: u16,
+        pub initial_credits: u16,
+        pub num_channels: u8,
+}
+
+fn to_l2cap_coc_connect(buffer: &[u8]) -> Result<CocConnectEvent, crate::event::Error> {
+    require_len!(buffer, 11);
```

Comment:
Shouldn't this be `13`?

Reply:
Answer is "no, because this is currently working so the length matches". But why do you think it should be 13 (and maybe by the time you answer, I can remember why I have it at 11...)?

Reply:
A few lines after 
```
num_channels: buffer[12] })
```
It works because 13 is more than 11, but if you would pass a 12 or 11 it would fail in the wrong spot.